neopixel_spi.py

```diff
@@ -91,45 +81,20 @@ class NeoPixel_SPI(NeoPixel):
         pixels = neopixel_spi.NeoPixel_SPI(board.SPI(), 10)
         pixels.fill(0xff0000)
     """
-    #pylint: disable=invalid-name, super-init-not-called, too-many-instance-attributes
+    #pylint: disable=invalid-name
 
     FREQ = 6400000  # 800kHz * 8, actual may be different
     TRST = 80e-6    # Reset code low level time
 
     def __init__(self, spi, n, *, bpp=3, brightness=1.0, auto_write=True, pixel_order=None):
-        # We can't call super().__init__() since we don't actually
-        # have a pin to supply it. So setup is done manually.
-        #
-        # neopixel stuff
-        #
-        self.bpp = bpp
-        self.n = n
+
+        # configure bpp and pixel_order
         if not pixel_order:
             pixel_order = GRB if bpp == 3 else GRBW
         else:
-            self.bpp = bpp = len(pixel_order)
-        #
-        # pypixelbuf stuff
-        #
-        bpp, byteorder_tuple, has_white, _ = self.parse_byteorder(pixel_order)
-        self._pixels = n
-        self._bytes = bpp * n
-        self._byteorder = byteorder_tuple
-        self._byteorder_string = pixel_order
-        self._has_white = has_white
-        self._bpp = bpp
-        self._bytearray = bytearray(n * bpp)
-        self._two_buffers = True
-        self._rawbytearray = bytearray(n * bpp)
-        self._offset = 0
-        self._dotstar_mode = False
-        self._pixel_step = bpp
-        self.auto_write = False
-        self.brightness = min(1.0, max(0, brightness))
-        self.auto_write = auto_write
-        #
-        # neopixel_spi stuff
-        #
+            bpp = len(pixel_order)
+
+        # set up SPI related stuff
         from adafruit_bus_device.spi_device import SPIDevice
```

Comment:
Please don't import from within a function. This can be at the top level now since it's been split from NeoPixel.

Reply:
yep. good catch.